pkg/sql/parser/sql.y

```diff
@@ -6135,9 +6139,26 @@ sortby_list:
   }
 
 sortby:
-  a_expr opt_asc_desc
+  a_expr opt_asc_desc opt_nulls_order
   {
-    $$.val = &tree.Order{OrderType: tree.OrderByColumn, Expr: $1.expr(), Direction: $2.dir()}
+    /* FORCE DOC */
+    dir := $2.dir()
+    nullsOrder := $3.nullsOrder()
+    // We currently only support the opposite of Postgres defaults.
+    if nullsOrder != tree.DefaultNullsOrder {
+      if dir == tree.Descending && nullsOrder == tree.NullsFirst {
+        return unimplementedWithIssue(sqllex, 6224)
+      }
+      if dir != tree.Descending && nullsOrder == tree.NullsLast {
+        return unimplementedWithIssue(sqllex, 6224)
```

Comment:
I think this will produce incorrect results with descending indexes, because `tree.DefaultDirection` is dependent on the index's direction.

Reply:
I'm not sure index direction matters here. A `SELECT ... ORDER BY` specifies the desired output ordering. If Postgres detects an index that happens to satisfy that ordering it will use it. There's no way in postgres to say `SELECT FROM t@idx ORDER BY blah` (postgres has no index hints).